jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPlugin.groovy

```diff
@@ -85,57 +85,23 @@ class JRubyPlugin implements Plugin<Project> {
         }
     }
 
-    // can not cast same object from different classloaders
-    private Object server;
-    private Object embeddedServer(Project project) {
-        if (server == null) {
-            // TODO maybe things will work now without cloning
-            // clone the current classloader without its parent
-            
-            // assume we run inside an URLClassLoader which might
-            // not always be the case (OSGi, J2EE, etc)
-            List<URL> urls = new LinkedList<URL>()
-            URL warFileURL
-            RubygemsServlet.class.getClassLoader().getURLs().each {
-                if (it.file.endsWith(".war") ) {
-                    warFileURL = it
-                }
-                // for integTest we need to filter some jars here
-                else if (!it.file.contains('bcprov-jdk15')) {
-                    urls.add(it)
-                }
-            }
-            ClassLoader extClassLoader = getClass().getClassLoader().getSystemClassLoader().parent
-            ClassLoader cl = new URLClassLoader(urls.toArray(new URL[urls.size()]), extClassLoader)
-            // can not cast
-            Object servlet = cl.loadClass(RubygemsServlet.class.getName())
-            server = servlet.create(warFileURL)
-        }
-        server
-    }
-
-    private boolean serverStarted = false
-    private void startEmbeddedServer(Project project) {
-        if (server != null && !serverStarted) {
-            // we need to set the current thread context class loader
-            // for starting up the webapps
-            ClassLoader current = Thread.currentThread().contextClassLoader
-            try {
-                Thread.currentThread().setContextClassLoader(server.class.classLoader)
-                if (project.logger.isDebugEnabled()) {
-                    server.enableLogging()
-                }
-                server.start()
-                serverStarted = true
-                project.gradle.buildFinished {
-                    server.stop()
-                    server.class.classLoader.close()
-                    server = null
-                }
-            }
-            finally {
-                Thread.currentThread().setContextClassLoader(current)
-            }
+    private void setupMavenGemProtocol(Project project) {
+        project.repositories {
+            jcenter()
```

Comment:
this can be tricky - the repository which contains the mavengem-protocol code needs to be first - maybe. the integration-tests worked with user supplied ```jcenter()``` but my test project failed until I but ```jcenter()``` as first repo. removed it for now.